src/shared/telemetry/spans.ts

```diff
@@ -159,16 +167,25 @@ export class TelemetrySpan {
     }
 }
 
+type Attributes = Partial<MetricShapes[MetricName]>
+
 interface TelemetryContext {
     readonly spans: TelemetrySpan[]
-    readonly activeSpan?: TelemetrySpan
+    readonly attributes: Attributes
 }
 
 // This class is called 'Telemetry' but really it can be used for any kind of tracing
 // You would need to make `Span` a template type and reduce the interface to just create/start/stop
 export class TelemetryTracer extends TelemetryBase {
     readonly #context = new AsyncLocalStorage<TelemetryContext>()
 
+    /**
+     * `record` may be called prior to entering any span. This field simulates the
+     * effect of having an ephemeral "root" span that only extends until we enter
```

Comment:
So this is like a "staging" area and can be used even before a span is entered?

Reply:
Yup. The ideal scenario is that all entry-points are covered by a span, but this isn't always practical.